### ISSUE TITLE
[HOLD for payment 2023-08-28] [Tracking Browser back button] [$1000] mWeb- Clicking on back button on Concierge after leaving the thread gives us access to the same thread again

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#20908]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to mweb chrome and go to any chat
2. Send a message and then click on reply thread on the message so that a new thread page opens up
3. Click on header and leave the room
4. Notice it shows you to Concierge page
5. Click on back button and notice that it brings the same thread chat again that you left few moments back. If a user has left the thread, clicking on back button should have brought the user to LHN instead of giving re-access to the same thread

## Expected Result:
Clicking on back button on Concierge after leaving the thread shouldn't give access to the same thread again (since the user had already decided to leave)

## Actual Result:
licking on back button on Concierge after leaving the thread gives us access to the same thread again

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.2.7.6

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/8aff1456-5b38-4116-8b07-011b45bcdade

https://github.com/Expensify/App/assets/93399543/cebc9a4e-5c5f-4b31-9603-863af4f38bdd

**Expensify/Expensify Issue URL:**

**Issue reported by:** @priya-zha

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686249319392879

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01448a7bf541bf3868</li>
        <li>Upwork Job ID: 1670845602095009792</li>
        <li>Last Price Increase: 2023-07-10</li>
    </ul>
</details>

**Comment by namhihi237** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Clicking on back button on Concierge after leaving the thread shouldn't give access to the same thread again (since the user had already decided to leave)

### What is the root cause of that problem?
The problem is that after pressing back it will return to the previous screen which is router `/r/reportID/details` because we haven't cleared the screen thread.

### What changes do you think we should make in order to solve the problem?
Define function pop screen in `Navigation.js`
```
function popScreens(numberScreen = 1) {
    navigationRef.current.dispatch(StackActions.pop(numberScreen));
}
```
In the function `leaveRoom` we will pop 2 screen before `navigateToConciergeChat`
```
Navigation.popScreens(2)
navigateToConciergeChat();
```
### What alternative solutions did you explore? (Optional)
N/A

**Comment by eh2077** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

On mWeb, clicking on back button on Concierge after leaving the thread gives us access to the same thread again

### What is the root cause of that problem?

The root cause of this issue is that, when clicking header go back button, the route state still saves previous pages(including the details page) after clicking `Leave Thread`, see [leaveRoom](https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/actions/Report.js#L1649), [navigateToConciergeChat](https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/actions/Report.js#L1132), and Header [Navigation.goBack](https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/pages/home/ReportScreen.js#L275).

### What changes do you think we should make in order to solve the problem?

From method [leaveRoom](https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/actions/Report.js#L1649)

https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/actions/Report.js#L1678

We know that we want to navigate to the concierge chat when clicking `Leave Thread` button. After navigating to the concierge chat page from the details page, we don't want to navigate back to the details page if clicking the header go back button. Instead, we want to go back to the home page.

To fix this issue, we can reset the route state of navigation to the concierge chat when calling method [leaveRoom](https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/actions/Report.js#L1649).

To achieve this, we can

1. Add a new type `RESET` support for method [Navigation.navigate](https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/Navigation/Navigation.js#L67)

https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/Navigation/Navigation.js#L62-L67

2. Change the [condition](https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/Navigation/linkTo.js#L43) of linkTo method to reset the state of navigation. If type is `RESET`, then reset the state like

```js
if (action !== undefined && type !== 'RESET') {
    root.dispatch(action);
} else {
    root.reset(state);
}
```

3. Add an optional parameter to method [navigateToConciergeChat](https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/actions/Report.js#L1132)

https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/actions/Report.js#L1132

like

```js
function navigateToConciergeChat(shouldResetNavigationState = false) {
```

If `shouldResetNavigationState` is true, then pass type `RESET` to method  [Navigation.navigate](https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/Navigation/Navigation.js#L67), like  `Navigation.navigate(route, 'RESET')`.

4. Call `navigateToConciergeChat(true);` in method leaveRoom

https://github.com/Expensify/App/blob/83ab0160a3b5c385fd26fffe20bdbccabf9249c0/src/libs/actions/Report.js#L1678

<details>
  <summary>Click to find the demo video</summary>

https://github.com/Expensify/App/assets/117511920/b6d80a79-cc8e-4c51-b77e-b43b703b22fc
</details>

### What alternative solutions did you explore? (Optional)

N/A

**Comment by abekkala** [comment]
@allroundexperts we already have a couple proposals for this one

**Comment by abekkala** [comment]
@allroundexperts have you had a chance to review the proposals that have come in?

**Comment by eh2077** [comment]
@allroundexperts Thanks for reviewing proposals. Can I have your feedback about my [proposal](https://github.com/Expensify/App/issues/20908#issuecomment-1595812584)? I proposed a solution to reset the state of navigation stack, which should be useful and complementary to the current navigating methods provided in [Navigation.js](https://github.com/Expensify/App/blob/main/src/libs/Navigation/Navigation.js). 

For example, we can open a page of App in a new tab through a link. If we perform an exit-and-redirect like navigation action, like leaving a thread, from that page, then having a reseting navigation stack solution will be helpful to avoid navigating to the `Hmm... it's not here` page in various use cases. 

cc @mountiny @WoLewicki

**Comment by WoLewicki** [comment]
Is this behavior the same on web? How does the forward browser button works in these flows if so? @eh2077 could you elaborate more on how would the `RESET` action API look like? It uses the `whole` state object as its argument (https://reactnavigation.org/docs/navigation-actions/#reset). How would we handle that in order to decide which part of the state we would keep etc?

The only problem with @namhihi237 proposal that I can think of is that it dispatches 2 navigation actions one after another, being popping the 2 screens and then navigating. I am afraid it can introduce some animation and browser buttons glitches. Normally resetting the state is the proper solution with removing the 2 last screens and adding another one at the end, but if you confirm that those mentioned things work as they should, I think we can also go with it.

**Comment by WoLewicki** [comment]
@namhihi237 and what happens if you click the forward browser button after leaving the thread? Seems weird that it is possible to even do that.

**Comment by abekkala** [comment]
Have we been able to reach a suitable proposal? 
@allroundexperts are you still waiting for input from @WoLewicki and @mountiny?

**Comment by abekkala** [comment]
@flaviadefaria Reassigning for Sabbatical

___

### CURRENT STATUS: 

- Issue Reported By: @priya-zha
- Proposal selected for fix: TBD
- C+ Review: @allroundexperts 

@allroundexperts should be getting to a conclusion on a proposal to accept **very soon** or take this to an expert contributor group to pick up!

**Comment by eh2077** [comment]
@allroundexperts Thanks for testing the proposal. Yes, I got same behavior.

**Comment by ShogunFire** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When we click on the back button in the concierge chat it renavigate to the thread we just closed

### What is the root cause of that problem?

The screen details and the thread report are still in the navigation stack

### What changes do you think we should make in order to solve the problem?

For the deatil page of the thread it is quite easy to remove we can just do a goBack() before navigating to concierge but we also need to remove the thread report from the stack

To do that we  can use reset, we go through all the report routes in the central panel route and we save them in a variable apart from the thread report one, then we pass those routes to the reset method. This is the method I have:

```
function popReportWithoutNavigation(reportId){
    var state = navigationRef.current.getRootState();
    //The route 1 is the CentralPanel
    var reportRoutes = state.routes[1].state.routes;
    
    const routes = [];
    reportRoutes.forEach(route => {
      if(route.params.reportID != reportId){
          routes.push(route);
      }
    });

    const resetAction = CommonActions.reset({
      index: 1,
      routes,
    });
    navigationRef.current.dispatch(resetAction);
}
```
we also can call that method before navigating to concierge chat








### What alternative solutions did you explore? (Optional)

**Comment by ShogunFire** [comment]
@WoLewicki @allroundexperts Hello, I think my proposal makes sense and it's works well on the back button but we still go back to the thread report when we press the forward button. I don't really know why, but could that be because the CentralPanelNavigator appears mutliple time at the root state ?

![image](https://github.com/Expensify/App/assets/19537677/3f058c22-fd86-4ed6-ab23-b2d3fc251fee)

**Comment by ShogunFire** [comment]
I don't really know how the browser has access to our stack

**Comment by flaviadefaria** [comment]
We're still waiting for new or improved proposals!

**Comment by mountiny** [comment]
Added this to the Navigation refactor follow up project so we can tackle this later if no good proposals arise

**Comment by mountiny** [comment]
Can you please link to the proposals and the issues? We can try to get them reviewed

**Comment by ShogunFire** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When we click on the back button in the concierge chat it renavigate to the thread we just closed

### What is the root cause of that problem?

The screen details and the thread report are still in the navigation stack.
So after we open the concierge chat the stack looks like this:

Home
CentralPaneNavigator: -> workspace report
CentralPaneNavigator -> thread report
RightModalNavigator `-> details page
CentralPaneNavigator -> concierge report

That's why when we go back we go back to the last screen in the stack which is details page

### What changes do you think we should make in order to solve the problem?


Before navigating to the concierge chat we want to get rid of:
CentralPaneNavigator -> thread report
RightModalNavigator `-> details page

I propose that we get rid of the thread report in the stack with the reset method and then we can call goBack or dismissModal to remove the details page.


To do the reset, we need to save all the routes but without the CentralPaneNavigator that contains the thread report, we can do it like this:

``` 
function popReportFromStack(reportId){

    const state = navigationRef.current.getRootState();
    
    const routesToKeep = [];
    state.routes.forEach(navigatorRoute => {
        if(navigatorRoute.name != NAVIGATORS.CENTRAL_PANE_NAVIGATOR){
            routesToKeep.push(navigatorRoute);
            return;
        }
            

        const reportRoutes = navigatorRoute.state.routes;
        
        var isNavigatorWeWantToDelete = false;
        reportRoutes.forEach(route => {
          if(route.params.reportID == reportId){
              isNavigatorWeWantToDelete = true;
          }
        });
        
        if(!isNavigatorWeWantToDelete){
            routesToKeep.push(navigatorRoute);
        }
    });
    const resetAction = CommonActions.reset({
      ...state,
      index: routesToKeep.length - 1,
      routes: routesToKeep,
    });
    navigationRef.current.dispatch(resetAction);
}
```




### What alternative solutions did you explore? (Optional)

**Comment by mountiny** [comment]
@adamgrzybowski @WoLewicki tagging you if you could please review the proposals here. thanks!

**Comment by mountiny** [comment]
Thanks! We will check it out. what issue are you referring to?

**Comment by ShogunFire** [comment]
Sorry my proposal is not complete :/ There is still a problem when we click forward from the concierge chat it goes to details... It looks like the browser doesn't use the state to know if he needs to put forward or back button.

Using goback will result in a forward button even if details is not in the stack, using navigate will result in a backward button to details even if details is not in the stack anymore :( 

@WoLewicki Do you have an idea how to resolve this problem ?

**Comment by ShogunFire** [comment]
@Nodebrute Have you seen my proposal ? I tried to use the reset, I had the right stack, but the problem is reset doesn't change the screen so at some point you either have to do pop push or replace.

pop => will show the forward button
push => will show the backward button and when you click it it goes to the screen you just left even if it's not in the stack
I haven't tried replace (it is probably the right way) because I read somewhere that you couldn't use it between navigators

Also we are kind of limited by the logic in linkTo, sometimes you try to do an option and it put another one

**Comment by ShogunFire** [comment]
Actually now that I think about it maybe we could navigate to concierge and then reset the stack, it not very hard to remove the right menu navigator, let me try

**Comment by ShogunFire** [comment]
> Let's consider a scenario where our navigation stack initially has four screens. When we remove one screen and replace it with the concierge screen, the total number of screens in the stack becomes three. This change in the stack can make the forward button active for one screen.

Sorry @Nodebrute  I hadn't understood your comment, you are right and it's happening even when we only do reset

I think not only do the browser saves the number of screens in the stack to know if he needs to show the forward button but he also saves the screen, for example:
Before: {1:Home, 2: Workspace, 3; thread, 4: Details } Index 4
After: {1: Home, 2: Workspace, 3: Concierge} Index 3

3<4 so he shows the forward button and 4 was the detail page so when we click on forward we go to the details page.

I am not sure how we can resolve this

**Comment by WoLewicki** [comment]
I am afraid it can be difficult to do it in a right way without some tricks. What if we did `pop` for as many screens as we need and then do a single `push` for the right screen on which we want to end on? We should be popping the screens till we end up on the one we want the user to go back to when clicking browser back button. I think in such cases when the navigation state would be kind of undefined after such actions, we should pop all the state till LHN and then just push the report we should end up on. 

Could you test what are the behaviors of browser buttons with such solution?

**Comment by ShogunFire** [comment]
I think that's what this proposal was about https://github.com/Expensify/App/issues/20908#issuecomment-1594948414 popping two screens and adding one. It stills shows the forward button :(

**Comment by WoLewicki** [comment]
Yeah, but `react-navigation` connects to the browser history API, so we have to look there.

**Comment by adamgrzybowski** [comment]
It's possible that the current way of handling history in the react-navigation isn't best suited for two actions one immediately after another. 

What react-navigation does for history is compare current and previous states of focused navigators and look for diff in routes length. (it's a bit simplified description)

For 3 x pop and one push the diff is still < 0 for routes array length so this action will be considered as go back in history

You can see the implementation here https://github.com/react-navigation/react-navigation/blob/24c03924397a6e59aba9f6b74a9c5cb4b939d9e1/packages/native/src/useLinking.tsx#L365-L378

If this problem cannot be solved in a different way maybe we need to look for a better implementation of handling history in react-navigation. But that may be tricky

**Comment by ShogunFire** [comment]
Thanks for the link, It looks like this code is not called twice otherwise the first time when we do pop(2) diff would be < 0 but then when we push diff would be > 0
I am going to check that when I have time

**Comment by lindboe** [comment]
> It's possible that the current way of handling history in the react-navigation isn't best suited for two actions one immediately after another.

This is true, here's an open issue: https://github.com/react-navigation/react-navigation/issues/10992.

I'm not familiar with the details, and from a quick skim it's not clear what the two actions being discussed here are, but if that's the core of the issue, it sounds like a potential solution might be a custom navigation option so that you can make all the desired updates in one go? https://reactnavigation.org/docs/custom-routers/#custom-navigation-actions

**Comment by flaviadefaria** [comment]
@allroundexperts please share your thoughts around [this](https://github.com/Expensify/App/issues/20908#issuecomment-1636361933) - thanks!

**Comment by flaviadefaria** [comment]
Thanks @allroundexperts! Let's take this internally as that's the action Melvin had already taken. I re-applied the engineering label to assign an engineer.

**Comment by hayata-suenaga** [comment]
so what is the bug here? is it that the user is taken to the concierge room when leaving a thread?

**Comment by flaviadefaria** [comment]
So the issues I see are two:
- The user does a thread in a room/DM and upon leaving the thread they are re-routed to Concierge instead of the original room/DM (ex:#announce room on the first video).
- After the user leaves the thread they can click the back button and are able to send a message in the thread despite having left it. Though now that I come to think of it, I'm not sure that this is a problem. Because leaving a thread means you're no longer following or getting notifications but it shouldn't necessarily prevent you from adding new messages to it if you'd like and being part of the thread again.

@hayata-suenaga maybe we should take this to Slack to confirm the desired outcome for this?

**Comment by hayata-suenaga** [comment]
will the second problem be fixed if we fix the first issue? Because the back button is on the Concierge page, if we don't direct users to the Concierge page after leaving a thread, we don't have to worry about the second issue?

**Comment by flaviadefaria** [comment]
Not really as I'd expect the back button to display in the original room. However, I'm not sure that the second issue is indeed an issue. I'll post in Slack and tag you.

**Comment by hayata-suenaga** [comment]
thank you so much for handing this issue @flaviadefaria 

by the way, I think this can be an external issue as this doesn't require backend change. I asked if anyone from Infinity Red is available to take over this issue [here](https://expensify.slack.com/archives/C04KQL2HSQ1/p1689787327880449)

**Comment by hayata-suenaga** [comment]
From this internal [slack discussion](https://expensify.slack.com/archives/C01SKUP7QR0/p1689785868129469), the expected behavior when pressing the back button (on small devices) is

taken back to whereenver the user was on before nativgating to the thread room. It might be LHN (users can directly access each thread room from LHN) or the room from which the thread was spawned.

**Comment by WoLewicki** [comment]
> The user does a thread in a room/DM and upon leaving the thread they are re-routed to Concierge instead of the original room/DM (ex:#announce room on the first video).

We should be able to change it rather easily.

As for both of those issues, I think the best solution for those problems is, after the user clicks `leave thread`, reset the navigation stack to only having LHN and the original report of the thread. At the same time, we should ensure that the browser stack is also reset to only those 2 elements on the stack. We will try and acquire it with the changes we try to implement in `react-navigation` with @adamgrzybowski. 
Right now I am not sure if we can do much about this issue. The best thing we could do is to pop the navigation state to the top and add the original report after that, but it will still probably create weird behaviors of browser buttons, as mentioned in this and other issues.

**Comment by mountiny** [comment]
@hayata-suenaga I recommend putting this issue on hold for the changes to the back button behaviour/ react-navigation changes. This issue is already tracked int he navigation refactor follow-ups project so it should be fine

**Comment by WoLewicki** [comment]
@hayata-suenaga we will start with making the changes that work correctly for the App's use cases. Testing those in the upstream repo is up to the maintainers of it, and those flows with browser navigation are usually kinda tricky, so it might take a while before it is done. Therefore I cannot give any reasonable estimation for it.

**Comment by hayata-suenaga** [comment]
per [this comment](https://github.com/Expensify/App/issues/20908#issuecomment-1644655756), putting this on hold

**Comment by mountiny** [comment]
@WoLewicki Which issue would you ties this to? We coul make it the main issue for it.

We dont have a tracking issue, there is a bunch with the same rootcause so we could find one which will be the main one but Wojti can help on that.

**Comment by WoLewicki** [comment]
Based on the number of comments, this one seems like the issue with most context 😅 All the other ones with similar root cause are: https://github.com/Expensify/App/issues/22101, https://github.com/Expensify/App/issues/21356, https://github.com/Expensify/App/issues/21214 (this one has many comments also).

**Comment by hayata-suenaga** [comment]
updated the title with these issues numbers so that I don't forget which issues I have to track to determine hold status of this   issue. Also, changing the priority to weekly

**Comment by mountiny** [comment]
@WoLewicki Lets use this one as the tracking issue then

**Comment by mountiny** [comment]
Updated the title to reflect we will use this one as the main one for the back button implementation

**Comment by flaviadefaria** [comment]
Is this still on hold or are we good to work on it now?

**Comment by WoLewicki** [comment]
Me and @adamgrzybowski are working on the changes in `react-navigation` that will fix those problems. Or are you proposing some other way to do it @flaviadefaria ?

**Comment by flaviadefaria** [comment]
Got it, thanks for clarifying. Should we switch this to monthly then until the changes are made in the library?

**Comment by hayata-suenaga** [comment]
yes probably that's a good idea changing the priority to monthly

**Comment by hayata-suenaga** [comment]
The same issue here: https://github.com/Expensify/App/issues/24123

**Comment by flaviadefaria** [comment]
@hayata-suenaga I think it is confusing to merge several issues into one. Each issue should have one reporting bonus and not have several linked to them. Or at least are you able to list in a single comment all the issues that are planned to be resolved here?

**Comment by hayata-suenaga** [comment]
@flaviadefaria I apologize for the confusion. I think this issue was originally an independent issue and was later used as a tracking issue.

I gonna ask Maria to create individual issues and put links to them here instead 🙇 

again sorry for the confusion

**Comment by adamgrzybowski** [comment]
Hey, this should be fixed by this [PR](https://github.com/Expensify/App/pull/24165)

**Comment by mountiny** [comment]
@flaviadefaria could you please confirm this issue is resolved in staging? If yes, then we can close it

**Comment by flaviadefaria** [comment]
My NewDot is really shitty lately and I can't log in at all today (this time also on web) and since I'm about to go OoO I'll reassign for someone else to test it.

**Comment by flaviadefaria** [comment]
@dylanexpensify (as you know) I'm unable to log in to NewDot and am about to go OoO. Can you please check if this bug is still happening as @mountiny asked above? Thanks!

**Comment by hayata-suenaga** [comment]
[The above issue](https://github.com/Expensify/App/issues/20908#issuecomment-1685937778) was [determined not a deploy blocker](https://github.com/Expensify/App/issues/25527#issuecomment-1686333833). Also, this issue was just mentioned in the above issue for reference and this issue didn't cause any regression (I mean the fix hasn't benn created yet).

**Comment by hayata-suenaga** [comment]
okay a lot of things going on here...

So there is no specific PR for this issue. A contributor [suggested](https://github.com/Expensify/App/issues/20908#issuecomment-1682260661) that [this PR](https://github.com/Expensify/App/pull/24165) is going to fix the issue reported here.

But [that PR apparently was reverted later](https://github.com/Expensify/App/pull/24165#issuecomment-1686339804)

We probably have to wait [until Vit reverts the revert](https://github.com/Expensify/App/pull/25572#issuecomment-1687091420). And then we can test if the issue is still reproducible.

@dylanexpensify Could you keep on eye on [the above comment](https://github.com/Expensify/App/pull/25572#issuecomment-1687091420) and test/reproduce the issue once the revert-revert PR is opened and merged?

I know this issue has a lot of comments and kind of confusing. Please let me know if you have any questions I can answer.

**Comment by mountiny** [comment]
The PR was not reverted, some people suggested it caused some deploy blocker but the revert did not confirm that (the issue prevailed) so we did not have to revert it (or in other words the revert branch pr was not merged)

This is fixed:

https://github.com/Expensify/App/assets/36083550/45d0d301-0a92-4589-8358-489736b2bf3f

@dylanexpensify I think we just need to pay the issue reporter and we can close this one out. No need to wait for the date in the title either honestly

**Comment by mountiny** [comment]
We are going to be more strict on these requests but given this issue has 114 comments and its been open since June 16 I think its valid to give the classic 25% so $250 for your help here. Albeit its not probably matching the efforts, there are other issues which end up being less time consuming so it balances out I believe. Thanks for your help

**Comment by hayata-suenaga** [comment]
@dylanexpensify so the issue was fixed in another PR. [The is the payment decision](https://github.com/Expensify/App/issues/20908#issuecomment-1687201122).

**Comment by dylanexpensify** [comment]
Got it! @allroundexperts @priya-zha mind applying [here](https://www.upwork.com/jobs/~01448a7bf541bf3868)

**Comment by priya-zha** [comment]
Hi. I'm not being able to apply to the job since it says 'The job is no longer available'

**Comment by dylanexpensify** [comment]
Woot! Payment summary:
- Reporter: @priya-zha $250
- Contributor+: @allroundexperts $250

[Upwork job](https://www.upwork.com/jobs/~01379dde4cb26bb333)! Please apply!

**Comment by dylanexpensify** [comment]
New job ^ fyi @priya-zha

**Comment by priya-zha** [comment]
Applied. Thanks

**Comment by JmillsExpensify** [comment]
Reviewed the details for @allroundexperts. $250 approved for payment via NewDot.

**Comment by dylanexpensify** [comment]
TY! Payment coming now @priya-zha

**Comment by dylanexpensify** [comment]
Offer sent @priya-zha !

**Comment by dylanexpensify** [comment]
bump @priya-zha

**Comment by priya-zha** [comment]
@dylanexpensify Accepted the offer already. Waiting for the payment. Thanks.

**Comment by flaviadefaria** [comment]
@dylanexpensify just to confirm have you paid @priya-zha? I see the contract is still open but it's unclear to me if they have been paid or not.

**Comment by dylanexpensify** [comment]
Payment sent!

**Comment by dylanexpensify** [comment]
Contract ended!

**Comment by flaviadefaria** [comment]
Perfect so closing this! Thanks!
